src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java

```diff
@@ -615,7 +615,14 @@ public Throwable getLastError() {
 			return this.ensureConnectTask(timeout)
 				.thenCompose(webSocket -> {
 					return sendAsyncLock.acquireThenCompose(timeout, () -> {
-						String json = command.getResponse().toJsonString();
+						String json = null;
+						if (command.getResponse() != null) {
+							json = command.getResponse().toJsonString();
+						}
+						if (command.getRenewToken() != null) {
+							json = command.getRenewToken().toJsonString();
+						}
```

Comment:
Logic like this could be made into a ListenerCommand.toJsonString() method. Or you could get rid of ListenerCommand all together and have each command implement an interface with a method like toJsonString() on it.

Reply:
I'm hoping that with the ListenerCommand changes this if/else-if will turn into a single line calling a method which hides any complexity like this.

---
In reply to: [267891127](https://github.com/Azure/azure-relay-java/pull/36#discussion_r267891127) [](ancestors = 267891127)